__tests__/commands/install/integration.js

```diff
@@ -124,6 +124,25 @@ test.concurrent(
   },
 );
 
+test.concurrent(
+  'resolves the symlinks of other symlinked packages relative to the package using the link: protocol',
+  async () => {
+    await runInstall({}, 'install-link-nested', async (config): Promise<void> => {
+      const expectPath = path.join(config.cwd, 'node_modules', 'b');
```

Comment:
Yes the following is what gets installed:
```
node_modules
  a (symlink) -> ../a
  b (symlink) -> ../a/b
```
This is however what is getting installed on develop currently:
```
node_modules
  a (symlink) -> ../a
  b (symlink) -> ../b
```
And that means the `b` symlink is going nowhere. I can't comment on deduping and that behavior as I'm new to the yarn codebase and have only basic understanding on resolvers in general and `ExoticResolver` logic specifically 😕

I can dig into it though and add a test case regarding:
> I wonder if several projects will link to a, would they be modifying their contents